lib/plugin.js

```diff
@@ -94,20 +94,16 @@ ManifestPlugin.prototype.apply = function(compiler) {
 
     var json = JSON.stringify(cache, null, 2);
 
-    compilation.assets[outputName] = {
-      source: function() {
-        return json;
-      },
-      size: function() {
-        return json.length;
-      }
-    };
-
-    if (this.opts.writeToFileEmit) {
-      var outputFolder = compilation.options.output.path;
-      var outputFile = path.join(outputFolder, this.opts.fileName);
+    var outputFolder = compilation.options.output.path;
+    var outputFile = path.join(outputFolder, this.opts.fileName);
 
+    const shouldUseNodeFs = compiler.outputFileSystem.constructor.name !== 'MemoryFileSystem' ||
```

Comment:
I know that `const` is much more appropriate here, but this file uses `var` all over the place (and even 2 previous lines use `var`)

So I guess we should stick to `var` and if we want to go with `const` we should create separate PR in which we do refactor all `var` to `const`